### PR TITLE
dev-java/opentest4j: add 1.3.0

### PR DESCRIPTION
**Pull request by vaukai**
---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

**Review by fordfrog — APPROVED** [review]

File: dev-java/opentest4j/opentest4j-1.3.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+RDEPEND=">=virtual/jre-1.8:*"
+
+JAVA_SRC_DIR=( src/{main,module}/java )
+
+JAVA_TEST_GENTOO_CLASSPATH="junit-4"
+JAVA_TEST_SRC_DIR=( src/test/java )
+JAVA_TEST_RESOURCE_DIRS=( src/test/resources )
+
+src_test() {
+	JAVAC_ARGS="--source 11 --target 11" java-pkg-simple_src_test
```
Comment (fordfrog) [inline dev-java/opentest4j/opentest4j-1.3.0.ebuild:31]: i guess `--target 11` makes it not runnable on `jre-1.8` as specified in `RDEPEND`. also it is not obvious here why it needs `jdk-11` for compilation, so i can't tell from this whether version 11 is also needed for runtime or runtime is fine with version 1.8.

**Review by fordfrog — CHANGES_REQUESTED** [review]

**Comment by fordfrog** [comment]
i merged the hamcrest commit. also, junit-5.11.0 is out afaik, so maybe it would be better to bump to that version instead.

File: dev-java/opentest4j/opentest4j-1.3.0.ebuild (hunk 1 above)
Comment (vaukai) [reply dev-java/opentest4j/opentest4j-1.3.0.ebuild:31]: > i guess `--target 11` makes it not runnable on `jre-1.8` as specified in `RDEPEND`. also it is not obvious here why it needs `jdk-11` for compilation, so i can't tell from this whether version 11 is also needed for runtime or runtime is fine with version 1.8.

we have `jdk-11` for compilation for `module-info.java` and runtime seems fine with `1.8`.
similar to `gson-2.11.0-r2` with `--source 17 --target 17`.

**Comment by vaukai** [comment]
> i merged the hamcrest commit. also, junit-5.11.0 is out afaik, so maybe it would be better to bump to that version instead.

i have `junit-5.11.0` wip but still compilation problems. nneds more time.

File: dev-java/opentest4j/opentest4j-1.3.0.ebuild (hunk 1 above)
Comment (fordfrog) [reply dev-java/opentest4j/opentest4j-1.3.0.ebuild:31]: well, according to documentation `--target 11` (or `--target 17`) generates bytecode of that java version, and such a bytecode cannot be run on a lower versions of java, for obvious reasons.

**Review by fordfrog — APPROVED** [review]

**Comment by fordfrog** [comment]
`junit-5.10.3` merged

**Pull request closed without merging**